Kudu.Contracts/Functions/IFunctionManager.cs

```diff
@@ -2,11 +2,20 @@
 using System.Collections.Generic;
 using System.Threading.Tasks;
 using Kudu.Core.SourceControl;
+using Kudu.Contracts.Jobs;
+using Newtonsoft.Json.Linq;
 
 namespace Kudu.Core.Functions
 {
     public interface IFunctionManager
     {
         Task SyncTriggers();
+        Task<FunctionEnvelope> CreateOrUpdate(string name, FunctionEnvelope functionEnvelope);
```

Comment:
Nit: Consider using the "Async" naming pattern for all the async functilons (e.g. SyncTriggersAsync, etc.)